### ISSUE TITLE
Teradata dialect: L019 incorrect identified in case when statements

### ISSUE THREAD

**Issue by its-gazza** [sqlfluff/sqlfluff#2492]
## Expected Behaviour

No changes to the commas as they're in the right place

## Observed Behaviour

The comma from the second `CASE WHEN` statement is transferred right after the first `CASE WHEN` statement's comma.

Rule `L019` incorrect identified the trailing comma in the second `CASE WHEN` statement as leading comma.

## Steps to Reproduce

Query in question:

```sql
SELECT
    Col1 AS Column1,
    CASE WHEN current_date = DATE '2021-01-01' THEN 1 ELSE 0 END AS Strt_Yr,
    CASE WHEN (Current_Date, Current_Date + 7) overlaps (DATE '2021-01-01', DATE '2021-01-07') THEN 1 ELSE NULL END AS col2,
    CASE WHEN (Current_Date, Current_Date + 7) overlaps (DATE '2021-01-01', DATE '2021-01-07') THEN 1 ELSE NULL END AS col3
```

Run `sql fix test.sql --force` will return

```sql
SELECT
    Col1 AS Column1,
    CASE WHEN current_date = DATE '2021-01-01' THEN 1 ELSE 0 END AS Strt_Yr, ,
    CASE WHEN (Current_Date, Current_Date + 7) overlaps (DATE '2021-01-01', DATE '2021-01-07') THEN 1 ELSE NULL END AS col2
    CASE WHEN (Current_Date, Current_Date + 7) overlaps (DATE '2021-01-01', DATE '2021-01-07') THEN 1 ELSE NULL END AS col3

```

## Dialect
`teradata`
## Version

```shell 
$ sqlfluff --version
sqlfluff, version 0.9.3

$ python --version
Python 3.9.9
```
Same error on python 3.6.8

## Configuration

```ini
[sqlfluff]
verbose = 2 
dialect = teradata
```

**Comment by barrywhart** [comment]
This query does not parse correctly. Unfortunately, `sqlfluff fix` will currently try to fix SQL that does not parse, leading to frequent issues like this. For now, the solution is to update the `teradata` dialect so the query will parse.

There is an open issue (#2424) about changing the behavior of `sqlfluff fix`. It's a high priority, but the maintainers need to discuss and agree before changing it.

**Comment by barrywhart** [comment]
Documentation for Teradata `OVERLAPS`:
https://docs.teradata.com/r/kmuOwjp1zEYg98JsB8fu_A/3VIgdwHNVU~tsnNiIR1aEw
`OVERLAPS` is part of ANSI 2011 SQL. See the "Filters" section of this page: https://illuminatedcomputing.com/posts/2019/08/sql2011-survey/.
